libultraship/libultraship/Lib/Fast3D/gfx_pc.cpp

```diff
@@ -1110,32 +1094,27 @@ static void gfx_sp_vertex(size_t n_vertices, size_t dest_index, const Vtx *verti
 
         // trivial clip rejection
         d->clip_rej = 0;
-        if (x < -w) d->clip_rej |= 1;
-        if (x > w) d->clip_rej |= 2;
-        if (y < -w) d->clip_rej |= 4;
-        if (y > w) d->clip_rej |= 8;
-        //if (z < -w) d->clip_rej |= 16;
-        if (z > w) d->clip_rej |= 32;
+        if (x < -w) d->clip_rej |= 1; // CLIP_LEFT
+        if (x > w) d->clip_rej |= 2; // CLIP_RIGHT
+        if (y < -w) d->clip_rej |= 4; // CLIP_BOTTOM
+        if (y > w) d->clip_rej |= 8; // CLIP_TOP
+        // if (z < -w) d->clip_rej |= 16; // CLIP_NEAR
+        if (z > w) d->clip_rej |= 32; // CLIP_FAR
 
         d->x = x;
         d->y = y;
         d->z = z;
         d->w = w;
 
         if (rsp.geometry_mode & G_FOG) {
-            if (fabsf(w) < 0.001f) {
-                // To avoid division by zero
-                w = 0.001f;
-            }
```

Comment:
This is a behavioral change. The old code converted all values in the range (-0.001, 0.001) to 0.001, but this converts all non-positive values to 0, which will certainly cause a divide by 0. I would suggest to leave this unchanged.

Reply:
totally missed that. reverted this now :)